fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/BestMvSelector.java

```diff
@@ -211,39 +554,82 @@ private int calcSortScore(MaterializedView mv, Set<String> equivalenceColumns, S
         return score;
     }
 
-    private CandidateContext getMVContext(
-            OptExpression expression, boolean isAggregate, OptimizerContext optimizerContext,
-            List<Table> diffTables, Set<String> equivalenceColumns, Set<String> nonEquivalenceColumns) {
-        if (expression.getOp() instanceof LogicalOlapScanOperator) {
-            LogicalOlapScanOperator scanOperator = expression.getOp().cast();
-            if (scanOperator.getTable().isMaterializedView()) {
-                MaterializedView mv = (MaterializedView) scanOperator.getTable();
-                if (!diffTables.contains(mv)) {
-                    return null;
-                }
-                int sortScore = calcSortScore(mv, equivalenceColumns, nonEquivalenceColumns);
-                CandidateContext candidateContext = new CandidateContext(
-                        expression.getStatistics(), scanOperator.getTable().getBaseSchema().size(), sortScore);
-                if (isAggregate) {
-                    List<MvPlanContext> planContexts = CachingMvPlanContextBuilder.getInstance().getPlanContext(
-                            optimizerContext.getSessionVariable(), mv);
-                    for (MvPlanContext planContext : planContexts) {
-                        if (planContext.getLogicalPlan().getOp() instanceof LogicalAggregationOperator) {
-                            LogicalAggregationOperator aggregationOperator = planContext.getLogicalPlan().getOp().cast();
-                            candidateContext.setGroupbyColumnNum(aggregationOperator.getGroupingKeys().size());
-                        }
-                    }
-                }
-                return candidateContext;
-            }
+    private boolean isColocateTable(Table table) {
+        if (table == null || !(table instanceof OlapTable)) {
+            return false;
+        }
+        OlapTable olapTable = (OlapTable) table;
+        if (!olapTable.isPartitionedTable()) {
+            return true;
```

Comment:
it seems that the unpartitioned table is not colocate if the table is distributed by random?